File: Android/app/src/androidTest/java/jp/co/riso/smartdeviceapp/model/PrinterTest.java

package jp.co.riso.smartdeviceapp.model;

import jp.co.riso.smartdeviceapp.AppConstants;
import jp.co.riso.smartdeviceapp.controller.printer.PrinterManager;
import jp.co.riso.smartdeviceapp.model.Printer.Config;
import jp.co.riso.smartdeviceapp.model.Printer.PortSetting;
import junit.framework.TestCase;
import android.os.Bundle;
import android.os.Parcel;

public class PrinterTest extends TestCase {
    final private String PRINTER_NAME = "Test Printer";
    final private String PRINTER_ADDRESS = "192.168.1.206";
    final private String PRINTER_TAG = "PRINTER_BUNDLE";
    final private String PRINTER_ARRAY_TAG = "PRINTER_ARRAY_BUNDLE";


    public PrinterTest(String name) {
        super(name);
    }

    protected void setUp() throws Exception {
        super.setUp();
    }

    protected void tearDown() throws Exception {
        super.tearDown();
    }

    // ================================================================================
    // Tests - constructors
    // ================================================================================

    public void testConstructor() {
        Printer printer = new Printer(PRINTER_NAME, PRINTER_ADDRESS);

        assertNotNull(printer);
    }
    
    public void testConstructor_Parcel() {
        Printer printer = new Printer(PRINTER_NAME, PRINTER_ADDRESS);
        Printer printerArray[] = new Printer[2];
        
        // Create parcelable object and put to Bundle
        Bundle bundlePut = new Bundle();

        // Null config
        printer.setConfig(null);
        // Parcel in
        bundlePut.putParcelable(PRINTER_TAG, printer);
        // New Array
        bundlePut.putParcelableArray(PRINTER_ARRAY_TAG, printerArray);
        
        // Save bundle to parcel
        Parcel parcel = Parcel.obtain();
        bundlePut.writeToParcel(parcel, 0);
        printer.writeToParcel(parcel, 0);
        
        parcel.setDataPosition(0);
        Bundle bundleExtract = parcel.readBundle();
        bundleExtract.setClassLoader(Printer.class.getClassLoader());
        printer = bundleExtract.getParcelable(PRINTER_TAG);
        bundleExtract.getParcelableArray(PRINTER_ARRAY_TAG);
        
        Printer createFromParcel = Printer.CREATOR.createFromParcel(parcel);
        parcel.recycle();
        assertNotNull(createFromParcel);
    }
    
    // ================================================================================
    // Tests - newArray
    // ================================================================================
    
    public void testNewArray_Parcel() {        
        Printer printer[] = Printer.CREATOR.newArray(2);
        assertNotNull(printer);
    }
    
    // ================================================================================
    // Tests - setId
    // ================================================================================

    public void testSetId() {
        try {
            Printer printer = new Printer("", "");
            int id = 0;

            printer.setId(id);
            assertEquals(id, printer.getId());
        } catch (Exception e) {
            fail(); // Error should not be thrown
        }
    }

    // ================================================================================
    // Tests - getId
    // ================================================================================

    public void testGetId_EmptyId() {
        try {
            Printer printer = new Printer("", "");

            assertEquals(PrinterManager.EMPTY_ID, printer.getId());
        } catch (Exception e) {
            fail(); // Error should not be thrown
        }
    }

    // ================================================================================
    // Tests - setName
    // ================================================================================

    public void testSetName_NullName() {
        try {
            Printer printer = new Printer("", "");

            printer.setName(null);
        } catch (Exception e) {
            fail(); // Error should not be thrown
        }
    }

    public void testSetName_ValidName() {
        try {
            Printer printer = new Printer("", "");

            printer.setName(PRINTER_NAME);
            assertEquals(PRINTER_NAME, printer.getName());
        } catch (Exception e) {
            fail(); // Error should not be thrown
        }
    }

    // ================================================================================
    // Tests - getIpAddress
    // ================================================================================

    public void testGetIpAddress() {
        try {
            Printer printer = new Printer("", "");

            printer.getIpAddress();
        } catch (Exception e) {
            fail(); // Error should not be thrown
        }
    }

    // ================================================================================
    // Tests - setIpAddress
    // ================================================================================

    public void testSetIpAddress_ValidAddress() {
        Printer printer = new Printer("", "");

        printer.setIpAddress(PRINTER_ADDRESS);
        assertEquals(PRINTER_ADDRESS, printer.getIpAddress());
    }

    public void testSetIpAddress_NullAddress() {
        try {
            Printer printer = new Printer("", "");

            printer.setIpAddress(null);
        } catch (Exception e) {
            fail(); // Error should not be thrown
        }
    }

    // ================================================================================
    // Tests - getPrinterType
    // ================================================================================

    public void testGetPrinterType_BlankName() {
        Printer printer = new Printer("", "");

        String printerType = printer.getPrinterType();
        assertEquals(printerType, AppConstants.PRINTER_MODEL_IS);
    }
    public void testGetPrinterType_IS() {
        Printer printer = new Printer("RISO IS950C-G", "");
        String printerType = printer.getPrinterType();
        assertEquals(printerType, AppConstants.PRINTER_MODEL_IS);

        printer = new Printer("RISO IS1000C-G", "");

        printerType = printer.getPrinterType();
        assertEquals(printerType, AppConstants.PRINTER_MODEL_IS);

        printer = new Printer("RISO IS1000C-J", "");

        printerType = printer.getPrinterType();
        assertEquals(printerType, AppConstants.PRINTER_MODEL_IS);
    }

    public void testGetPrinterType_GD() {
        Printer printer = new Printer("RISO IS950C-GD", "");
        String printerType = printer.getPrinterType();
        assertEquals(printerType, AppConstants.PRINTER_MODEL_GD);

        printer = new Printer("ORPHIS GD", "");

        printerType = printer.getPrinterType();
        assertEquals(printerType, AppConstants.PRINTER_MODEL_GD);

        printer = new Printer("ORPHIS GDMODEL", "");

        printerType = printer.getPrinterType();
        assertEquals(printerType, AppConstants.PRINTER_MODEL_GD);
    }

    public void testGetPrinterType_FW() {
        Printer printer = new Printer("RISO IS950C-FW", "");
        String printerType = printer.getPrinterType();
        assertEquals(printerType, AppConstants.PRINTER_MODEL_FW);

        printer = new Printer("ORPHIS FW", "");

        printerType = printer.getPrinterType();
        assertEquals(printerType, AppConstants.PRINTER_MODEL_FW);

        printer = new Printer("ORPHIS FWMODEL", "");

        printerType = printer.getPrinterType();
        assertEquals(printerType, AppConstants.PRINTER_MODEL_FW);
    }

    // ================================================================================
    // Tests - getPortSetting
    // ================================================================================

    public void testGetPortSetting() {
        try {
            Printer printer = new Printer("", "");

            printer.getPortSetting();
        } catch (Exception e) {
            fail(); // Error should not be thrown
        }
    }

    // ================================================================================
    // Tests - setPortSetting
    // ================================================================================

    public void testSetPortSetting() {
        try {
            Printer printer = new Printer("", "");
            int portSetting = 1;

            printer.setPortSetting(PortSetting.RAW);
            assertTrue(PortSetting.RAW.equals(printer.getPortSetting()));
            assertEquals(PortSetting.RAW.ordinal(), portSetting);

            portSetting = 0;
            printer.setPortSetting(PortSetting.LPR);
            assertTrue(PortSetting.LPR.equals(printer.getPortSetting()));
            assertEquals(PortSetting.LPR.ordinal(), portSetting);
        } catch (Exception e) {
            fail(); // Error should not be thrown
        }
    }

    // ================================================================================
    // Tests - getConfig
    // ================================================================================

    public void testGetConfig() {
        Printer printer = new Printer("", "");
        Config config = printer.getConfig();

        assertNotNull(config);
    }

    // ================================================================================
    // Tests - setConfig
    // ================================================================================

    public void testSetConfig_ValidConfig() {
        Printer printer = new Printer("", "");
        Config config = printer.new Config();

        printer.setConfig(config);
        assertNotNull(config);
    }

    public void testSetConfig_NullConfig() {
        try {
            Printer printer = new Printer("", "");

            printer.setConfig(null);
        } catch (Exception e) {
            fail(); // Error should not be thrown
        }
    }
    
    // ================================================================================
    // Tests - describeContents
    // ================================================================================

    public void testDescribeContents() {
        Printer printer = new Printer("", "");
        printer.describeContents();
    }
    
    // ================================================================================
    // Tests - Config
    // ================================================================================

    public void testConfig() {
        Printer printer = new Printer("", "");
        printer.getConfig().isLprAvailable();
        printer.getConfig().isRawAvailable();
        printer.getConfig().isBookletFinishingAvailable();
        printer.getConfig().isStaplerAvailable();
        printer.getConfig().isPunch3Available();
        printer.getConfig().isPunch4Available();
        printer.getConfig().isTrayFaceDownAvailable();
        printer.getConfig().isTrayTopAvailable();
        printer.getConfig().isTrayStackAvailable();
        printer.getConfig().isTrayTopAvailable();
    }
}

File: Android/app/src/androidTest/kotlin/jp/co/riso/smartdeviceapp/model/PrinterTest.kt
package jp.co.riso.smartdeviceapp.model

import android.os.Bundle
import android.os.Parcel
import jp.co.riso.smartdeviceapp.AppConstants
import jp.co.riso.smartdeviceapp.controller.printer.PrinterManager
import jp.co.riso.smartdeviceapp.model.Printer.PortSetting
import junit.framework.TestCase

class PrinterTest(name: String?) : TestCase(name) {
    private val PRINTER_NAME = "Test Printer"
    private val PRINTER_ADDRESS = "192.168.1.206"
    private val PRINTER_TAG = "PRINTER_BUNDLE"
    private val PRINTER_ARRAY_TAG = "PRINTER_ARRAY_BUNDLE"

    @Throws(Exception::class)
    override fun setUp() {
        super.setUp()
    }

    @Throws(Exception::class)
    override fun tearDown() {
        super.tearDown()
    }

    // ================================================================================
    // Tests - constructors
    // ================================================================================
    fun testConstructor() {
        val printer = Printer(PRINTER_NAME, PRINTER_ADDRESS)
        assertNotNull(printer)
    }

    fun testConstructor_NullName() {
        val printer = Printer(null, PRINTER_ADDRESS)
        assertNotNull(printer)
    }

    fun testConstructor_Parcel() {
        val printer = Printer(PRINTER_NAME, PRINTER_ADDRESS)
        val printerArray = arrayOfNulls<Printer>(2)

        // Create parcelable object and put to Bundle
        val bundlePut = Bundle()

        // Null config
        printer.config = null
        // Parcel in
        bundlePut.putParcelable(PRINTER_TAG, printer)
        // New Array
        bundlePut.putParcelableArray(PRINTER_ARRAY_TAG, printerArray)

        // Save bundle to parcel
        val parcel = Parcel.obtain()
        bundlePut.writeToParcel(parcel, 0)
        printer.writeToParcel(parcel, 0)
        parcel.setDataPosition(0)
        val bundleExtract = parcel.readBundle()
        bundleExtract!!.classLoader = Printer::class.java.classLoader
        bundleExtract.getParcelableArray(PRINTER_ARRAY_TAG)
        val createFromParcel = Printer.CREATOR.createFromParcel(parcel)!!
        parcel.recycle()
        assertNotNull(createFromParcel)
    }

    // ================================================================================
    // Tests - newArray
    // ================================================================================
    fun testNewArray_Parcel() {
        val printer: Array<out Printer?>? = Printer.CREATOR.newArray(2)
        assertNotNull(printer)
    }

    // ================================================================================
    // Tests - setId
    // ================================================================================
    fun testSetId() {
        try {
            val printer = Printer("", "")
            val id = 0
            printer.id = id
            assertEquals(id, printer.id)
        } catch (e: Exception) {
            fail() // Error should not be thrown
        }
    }

    // ================================================================================
    // Tests - getId
    // ================================================================================
    fun testGetId_EmptyId() {
        try {
            val printer = Printer("", "")
            assertEquals(PrinterManager.EMPTY_ID, printer.id)
        } catch (e: Exception) {
            fail() // Error should not be thrown
        }
    }

    // ================================================================================
    // Tests - setName
    // ================================================================================
    fun testSetName_NullName() {
        try {
            val printer = Printer("", "")
            printer.name = null
        } catch (e: Exception) {
            fail() // Error should not be thrown
        }
    }

    fun testSetName_ValidName() {
        try {
            val printer = Printer("", "")
            printer.name = PRINTER_NAME
            assertEquals(PRINTER_NAME, printer.name)
        } catch (e: Exception) {
            fail() // Error should not be thrown
        }
    }

    // ================================================================================
    // Tests - getIpAddress
    // ================================================================================
    fun testGetIpAddress() {
        try {
            val printer = Printer("", "")
            printer.ipAddress
        } catch (e: Exception) {
            fail() // Error should not be thrown
        }
    }

    // ================================================================================
    // Tests - setIpAddress
    // ================================================================================
    fun testSetIpAddress_ValidAddress() {
        val printer = Printer("", "")
        printer.ipAddress = PRINTER_ADDRESS
        assertEquals(PRINTER_ADDRESS, printer.ipAddress)
    }

    fun testSetIpAddress_NullAddress() {
        try {
            val printer = Printer("", "")
            printer.ipAddress = null
        } catch (e: Exception) {
            fail() // Error should not be thrown
        }
    }

    // ================================================================================
    // Tests - getPrinterType
    // ================================================================================
    fun testGetPrinterType_BlankName() {
        val printer = Printer("", "")
        val printerType = printer.printerType
        assertEquals(printerType, AppConstants.PRINTER_MODEL_IS)
    }

    fun testGetPrinterType_IS() {
        var printer = Printer("RISO IS950C-G", "")
        var printerType = printer.printerType
        assertEquals(printerType, AppConstants.PRINTER_MODEL_IS)
        printer = Printer("RISO IS1000C-G", "")
        printerType = printer.printerType
        assertEquals(printerType, AppConstants.PRINTER_MODEL_IS)
        printer = Printer("RISO IS1000C-J", "")
        printerType = printer.printerType
        assertEquals(printerType, AppConstants.PRINTER_MODEL_IS)
        printer = Printer("RISO IS10", "")
        printerType = printer.printerType
        assertEquals(printerType, AppConstants.PRINTER_MODEL_IS)
    }

    fun testGetPrinterType_GD() {
        var printer = Printer("RISO IS950C-GD", "")
        var printerType = printer.printerType
        assertEquals(printerType, AppConstants.PRINTER_MODEL_GD)
        printer = Printer("ORPHIS GD", "")
        printerType = printer.printerType
        assertEquals(printerType, AppConstants.PRINTER_MODEL_GD)
        printer = Printer("ORPHIS GDMODEL", "")
        printerType = printer.printerType
        assertEquals(printerType, AppConstants.PRINTER_MODEL_GD)
    }

    fun testGetPrinterType_FW() {
        var printer = Printer("RISO IS950C-FW", "")
        var printerType = printer.printerType
        assertEquals(printerType, AppConstants.PRINTER_MODEL_FW)
        printer = Printer("ORPHIS FW", "")
        printerType = printer.printerType
        assertEquals(printerType, AppConstants.PRINTER_MODEL_FW)
        printer = Printer("ORPHIS FWMODEL", "")
        printerType = printer.printerType
        assertEquals(printerType, AppConstants.PRINTER_MODEL_FW)
    }

    fun testGetPrinterType_FT() {
        var printer = Printer("RISO IS950C-FT", "")
        var printerType = printer.printerType
        assertEquals(printerType, AppConstants.PRINTER_MODEL_FT)
        printer = Printer("ORPHIS FT", "")
        printerType = printer.printerType
        assertEquals(printerType, AppConstants.PRINTER_MODEL_FT)
        printer = Printer("ComColor FT", "")
        printerType = printer.printerType
        assertEquals(printerType, AppConstants.PRINTER_MODEL_FT)
    }

    fun testGetPrinterType_GL() {
        var printer = Printer("RISO IS950C-GL", "")
        var printerType = printer.printerType
        assertEquals(printerType, AppConstants.PRINTER_MODEL_GL)
        printer = Printer("ORPHIS GL", "")
        printerType = printer.printerType
        assertEquals(printerType, AppConstants.PRINTER_MODEL_GL)
        printer = Printer("ComColor GL", "")
        printerType = printer.printerType
        assertEquals(printerType, AppConstants.PRINTER_MODEL_GL)
    }

    fun testGetPrinterType_CEREZONA_S() {
        var printer = Printer("RISO CEREZONA S", "")
        var printerType = printer.printerType
        assertEquals(printerType, AppConstants.PRINTER_MODEL_FT)
        printer = Printer("RISO CEREZONA S200", "")
        printerType = printer.printerType
        assertEquals(printerType, AppConstants.PRINTER_MODEL_FT)
    }

    // ================================================================================
    // Tests - getPortSetting
    // ================================================================================
    fun testGetPortSetting() {
        try {
            val printer = Printer("", "")
            printer.portSetting
        } catch (e: Exception) {
            fail() // Error should not be thrown
        }
    }

    // ================================================================================
    // Tests - setPortSetting
    // ================================================================================
    fun testSetPortSetting() {
        try {
            val printer = Printer("", "")
            var portSetting = 1
            printer.portSetting = PortSetting.RAW
            assertTrue(PortSetting.RAW == printer.portSetting)
            assertEquals(PortSetting.RAW.ordinal, portSetting)
            portSetting = 0
            printer.portSetting = PortSetting.LPR
            assertTrue(PortSetting.LPR == printer.portSetting)
            assertEquals(PortSetting.LPR.ordinal, portSetting)
        } catch (e: Exception) {
            fail() // Error should not be thrown
        }
    }

    // ================================================================================
    // Tests - getConfig
    // ================================================================================
    fun testGetConfig() {
        val printer = Printer("", "")
        val config = printer.config
        assertNotNull(config)
    }

    // ================================================================================
    // Tests - setConfig
    // ================================================================================
    fun testSetConfig_ValidConfig() {
        val printer = Printer("", "")
        val config = printer.Config()
        printer.config = config
        assertNotNull(config)
    }

    fun testSetConfig_NullConfig() {
        try {
            val printer = Printer("", "")
            printer.config = null
        } catch (e: Exception) {
            fail() // Error should not be thrown
        }
    }

    // ================================================================================
    // Tests - describeContents
    // ================================================================================
    fun testDescribeContents() {
        val printer = Printer("", "")
        printer.describeContents()
    }

    // ================================================================================
    // Tests - Config
    // ================================================================================
    fun testGetConfigParams() {
        val printer = Printer("", "")
        printer.config!!.isLprAvailable
        printer.config!!.isRawAvailable
        printer.config!!.isBookletFinishingAvailable
        printer.config!!.isStaplerAvailable
        printer.config!!.isPunch3Available
        printer.config!!.isPunch4Available
        printer.config!!.isTrayFaceDownAvailable
        printer.config!!.isTrayTopAvailable
        printer.config!!.isTrayStackAvailable
        printer.config!!.isTrayTopAvailable
        printer.config!!.isExternalFeederAvailable
        printer.config!!.isPunch0Available
    }

    fun testSetConfigParams() {
        val printer = Printer("", "")
        printer.config!!.isLprAvailable = true
        printer.config!!.isRawAvailable = true
        printer.config!!.isBookletFinishingAvailable = true
        printer.config!!.isStaplerAvailable = true
        printer.config!!.isPunch3Available = true
        printer.config!!.isPunch4Available = true
        printer.config!!.isTrayFaceDownAvailable = true
        printer.config!!.isTrayTopAvailable = true
        printer.config!!.isTrayStackAvailable = true
        printer.config!!.isTrayTopAvailable = true
        printer.config!!.isExternalFeederAvailable = true
        printer.config!!.isPunch0Available = true
    }

    fun testGetPunchAvailability() {
        val printer = Printer("", "")
        printer.config!!.isPunchAvailable

        // isPunch0Available is true
        printer.config!!.isPunch0Available = true
        printer.config!!.isPunch3Available = false
        printer.config!!.isPunch4Available = false
        printer.config!!.isPunchAvailable

        printer.config!!.isPunch0Available = true
        printer.config!!.isPunch3Available = true
        printer.config!!.isPunch4Available = false
        printer.config!!.isPunchAvailable

        printer.config!!.isPunch0Available = true
        printer.config!!.isPunch3Available = false
        printer.config!!.isPunch4Available = true
        printer.config!!.isPunchAvailable

        printer.config!!.isPunch0Available = true
        printer.config!!.isPunch3Available = true
        printer.config!!.isPunch4Available = true
        printer.config!!.isPunchAvailable

        // isPunch0Available is false
        printer.config!!.isPunch0Available = false
        printer.config!!.isPunch3Available = false
        printer.config!!.isPunch4Available = false
        printer.config!!.isPunchAvailable

        printer.config!!.isPunch0Available = false
        printer.config!!.isPunch3Available = true
        printer.config!!.isPunch4Available = false
        printer.config!!.isPunchAvailable

        printer.config!!.isPunch0Available = false
        printer.config!!.isPunch3Available = false
        printer.config!!.isPunch4Available = true
        printer.config!!.isPunchAvailable

        printer.config!!.isPunch0Available = false
        printer.config!!.isPunch3Available = true
        printer.config!!.isPunch4Available = true
        printer.config!!.isPunchAvailable
    }
}